src/scripts/api.js

```
@@ -140,255 +1,341 @@
+                resolve(this.sample);
+            } else if (!this.identifier || !this.path) {
+                reject();
+            } else if (magisterApi.cache[this.identifier] && !magisterApi.cache[this.identifier].then) {
+                resolve(magisterApi.cache[this.identifier]);
+            } else {
+                let res = await this.#fetchWrapper(
+                    `https://${magisterApi.schoolName}.magister.net/${this.path}`,
+                );
+                resolve(res);
+            }
+        });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)